java-client/build.gradle.kts

```diff
@@ -169,6 +169,7 @@ dependencies {
     val jacksonDatabindVersion = "2.17.0"
 
     // Apache 2.0
+    api("commons-logging:commons-logging:1.3.2")
```

Comment:
This dependency is coming from `opensearch-rest-client` transitively, making it a mandatory one since it is also needed by Apache HttpClient 5 transport.